src/components/parser/compilerlog.ts

```diff
@@ -114,11 +114,35 @@ export class CompilerLogParser {
         return false
     }
 
-    showCompilerDiagnostics(compilerDiagnostics: vscode.DiagnosticCollection, buildLog: LogEntry[], source: string) {
+    async showCompilerDiagnostics(compilerDiagnostics: vscode.DiagnosticCollection, buildLog: LogEntry[], source: string) {
         compilerDiagnostics.clear()
         const diagsCollection = Object.create(null) as { [key: string]: vscode.Diagnostic[] }
         for (const item of buildLog) {
-            const range = new vscode.Range(new vscode.Position(item.line - 1, 0), new vscode.Position(item.line - 1, 65535))
+            let startChar = 0
+            let endChar = 65535
+            if (item.errorPosText) {
+                // try to find the errorPosText in the respective line of the document
+                try {
+                    const document = await vscode.workspace.openTextDocument(item.file)
```

Comment:
This opens `item.file` in a new editor. I think some users will find it annoying. I suggest to read the content of this file by using https://github.com/James-Yu/LaTeX-Workshop/blob/e00be44387417c23c54bedc7897f85f780bc59ca/src/components/manager.ts#L515-L518

Note that for this to work you have
- to add `this.extension = extension` to the `constructor` and use `this.extension.manager.getDirtyContent`
- `item.file` must be a full path

Reply:
I tested this and `openTextDocument` does not actually open the file in the UI. There is a separate method `vscode.window.showTextDocument` for showing the document in the UI.

I applied your suggestion anyway, but I guess the openTextDocument method has faster access to the string of a line, so maybe you want to merge without the last commit.

Reply:
I tested this and openTextDocument does not actually open the file in the UI. There is a separate method vscode.window.showTextDocument for showing the document in the UI.

I applied your suggestion anyway, but I guess the openTextDocument method has faster access to the string of a line, so maybe you want to merge without the last commit.